src/sciline/_provider.py

```diff
@@ -3,19 +3,27 @@
 from __future__ import annotations
 
 import inspect
+from dataclasses import dataclass
 from typing import (
     TYPE_CHECKING,
     Any,
     Callable,
     Generator,
+    Literal,
+    Optional,
     TypeVar,
+    Union,
     get_args,
     get_origin,
     get_type_hints,
 )
 
 if TYPE_CHECKING:
-    from .typing import Key, Provider
+    from .typing import Key
+
+
+ToProvider = Callable[..., Any]
+ProviderKind = Literal['function', 'parameter', 'series', 'table', 'sentinel']
```

Comment:
Any reason why we use literals instead of, e.g., an enum?

Reply:
It was like this and I didn't want to change too much. I don't mind either way